server.js

```diff
@@ -21,11 +24,20 @@ app.use((err, req, res, next) => {
     res.status(404).send({ msg: "Article not found" });
   } else if (err.msg === "Comments not found") {
     res.status(404).send({ msg: "Comments not found" });
-  } else next();
+  } else next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.detail.includes("Key (article_id)=")) {
+    res.status(404).send({ msg: "Article not found" });
+  } else if (err.code === "23502" || err.code === "23503") {
+    res.status(400).send({ msg: "Bad request" });
+  } else next(err);
```

Comment:
This is an interesting way of checking for the error, but at a glance this is not very obvious. I would recommend saving the first if statement condition to a variable that is descriptive and shows what you are actually checking for. I.e. `hasArticleError`